app/Model/Cou.php

```diff
@@ -259,40 +259,73 @@ public function childCous($parentCou, $co_id, $includeParent=false) {
    * @since  COmanage Registry v2.0.0
    * @param  Integer $parentCouId   COU ID for Parent COU
    * @param  Boolean $includeParent Whether or not to return $parentCou in the results
+   * @param  Object  $parent        Whether or not parent object exists
    * @return Array List of COU IDs and Names
-   * @throws InvalidArgumentException
    */
   
-  public function childCousById($parentCouId, $includeParent=false) {
-    // Find $parentCou
-    
-    $args = array();
-    $args['conditions']['Cou.id'] = $parentCouId;
-    $args['contain'] = false;
-    
-    $parent = $this->find('first', $args);
-    
-    if(!isset($parent['Cou']['id'])) {
-      throw new InvalidArgumentException(_txt('er.unknown', array($parentCouId)));
+  public function childCousById($parentCouId, $includeParent=false, $parent=NULL) {
+    // Find $parent if not exists
+    if(empty($parent)) { 
```

Comment:
Why not even shorter?
$parent = empty($parent) ? $this->getParentCouById($parentCouId) : $parent;